RecoJets/JetProducers/src/MVAJetPuId.cc

```diff
@@ -0,0 +1,568 @@
+		ptRMS/=internalId_.nTot_;
+		ptRMS=sqrt(ptRMS);
+		internalId_.jetRchg_ = internalId_.leadChPt_/sumPt;
+		internalId_.jetR_ = internalId_.leadPt_/sumPt;
+
+		internalId_.ptMean_  = ptMean;
+		internalId_.ptRMS_   = ptRMS/jetPt;
+		internalId_.pt2A_    = sqrt( internalId_.ptD_     /internalId_.nTot_)/jetPt;
+		internalId_.ptD_     = sqrt( internalId_.ptD_)    / sumPt;
+		internalId_.sumPt_   = sumPt;
+		internalId_.sumChPt_ = sumPtCh;
+		internalId_.sumNePt_ = sumPtNe;
+		if (sumPt > 0) {
+			internalId_.beta_     /= sumPt;
+			internalId_.betaStar_ /= sumPt;
+		} else {
+			assert( internalId_.beta_ == 0. && internalId_.betaStar_ == 0.);
+		}
+		float ave_deta = sum_deta/sumPt2;
+		float ave_dphi = sum_dphi/sumPt2;
+		float ave_deta2 = sum_deta2/sumPt2;
+		float ave_dphi2 = sum_dphi2/sumPt2;
+		float a = ave_deta2-ave_deta*ave_deta;
+		float b = ave_dphi2-ave_dphi*ave_dphi;
+		float c = -(sum_detadphi/sumPt2-ave_deta*ave_dphi);
+		float delta = sqrt(fabs((a-b)*(a-b)+4*c*c));
+		float axis1=0; float axis2=0;
+		if (a+b+delta > 0) {
+			axis1 = sqrt(0.5*(a+b+delta));
+		}
+		if (a+b-delta > 0) {
+			axis2 = sqrt(0.5*(a+b-delta));
+		}
+		internalId_.axisMajor_ = axis1; //sqrt(fabs(eigVals(0)));
+		internalId_.axisMinor_ = axis2;//sqrt(fabs(eigVals(1)));
+		//compute Pull
+
+		float ddetaR_sum(0.0), ddphiR_sum(0.0);
+		for(int i=0; i<internalId_.nTot_; ++i) {
+			reco::PFCandidatePtr part = pfjet->getPFConstituent(i);
+			float weight =part->pt()*part->pt() ;
+			float deta = part->eta() - jet->eta();
+			float dphi = 2*atan(tan((part->phi()-jet->phi())/2));
+			float ddeta, ddphi, ddR;
+			ddeta = deta - ave_deta ;
+			ddphi = 2*atan(tan((dphi - ave_dphi)/2.)) ;
+			ddR = sqrt(ddeta*ddeta + ddphi*ddphi);
+			ddetaR_sum += ddR*ddeta*weight;
+			ddphiR_sum += ddR*ddphi*weight;
+		}if (sumPt2 > 0) {
+			float ddetaR_ave = ddetaR_sum/sumPt2;
+			float ddphiR_ave = ddphiR_sum/sumPt2;
+			internalId_.dRMean_ = sqrt(ddetaR_ave*ddetaR_ave+ddphiR_ave*ddphiR_ave);
+		}
+
+
+
+	}
+
+	if( calculateMva ) {
+		runMva();
+	}
+
+	return PileupJetIdentifier(internalId_);
+}
+
+std::string MVAJetPuId::dumpVariables() const
+{
+	std::stringstream out;
+	for(variables_list_t::const_iterator it=variables_.begin(); 
+			it!=variables_.end(); ++it ) {
+		out << std::setw(15) << it->first << std::setw(3) << "=" 
+			<< std::setw(5) << *it->second.first 
+			<< " (" << std::setw(5) << it->second.second << ")" << std::endl;
+	}
+	return out.str();
+}
+
+void MVAJetPuId::resetVariables()
+{
+	internalId_.idFlag_    = 0;
+	for(variables_list_t::iterator it=variables_.begin(); 
+			it!=variables_.end(); ++it ) {
+		*it->second.first = it->second.second;
+	}
+}
+
+#define INIT_VARIABLE(NAME,TMVANAME,VAL)    \
+	internalId_.NAME ## _ = VAL; \
+variables_[ # NAME   ] = std::make_pair(& internalId_.NAME ## _, VAL);
+
+void MVAJetPuId::initVariables()
+{
+	internalId_.idFlag_    = 0;
```

Comment:
why are these defaults set here instead of the PileupJetIdentifier default constructor ?